start_esp/nginx-auto.conf.template

```diff
@@ -115,19 +115,42 @@ http {
 
 % if cors_preset == 'basic':
     # Begin basic CORS preset
-    if (${'$'}request_method = 'OPTIONS') {
-      add_header 'Access-Control-Max-Age' 1728000;
-      add_header 'Access-Control-Allow-Origin' '${cors_allow_origin}';
-      add_header 'Access-Control-Allow-Methods' '${cors_allow_methods}';
-      add_header 'Access-Control-Allow-Headers' '${cors_allow_headers}';
-      add_header 'Content-Type' 'text/plain; charset=utf-8';
-      add_header 'Content-Length' 0;
-      return 204;
+  % if cors_allow_origin_regex != '':
+    if ($http_origin ~* (${cors_allow_origin_regex})) {
+        set $cors "true";
+        set $allow_origin $http_origin;
+    }
+  % else:
+    set $cors "true";
+    set $allow_origin "${cors_allow_origin}";
+  % endif
+
+    if ($request_method = 'OPTIONS') {
+        set $cors "${'$'}{cors}options";
+    }
+
+    if ($cors = "trueoptions") {
+        add_header 'Access-Control-Max-Age' 1728000;
+        add_header 'Access-Control-Allow-Origin' "$allow_origin";
+        add_header 'Access-Control-Allow-Methods' '${cors_allow_methods}';
+        add_header 'Access-Control-Allow-Headers' '${cors_allow_headers}';
+      % if cors_allow_credentials and cors_allow_origin != '*':
```

Comment:
this may have problem:   cors_allow_origin default to "*".   If a user is using cors_allow_origin_regex,  he would not be able to turn on allow_credentail  if he doesn't change cors_allow_origin flag.